File: 1-Easy/14. Longest Common Prefix/Horizontal_scanning.py
class Solution:
    def longestCommonPrefix(self, strs):
        if not strs:
            return ""
        pre = list(strs[0])
        for i in range(1, len(strs)):
            try:
                while strs[i].index(strs[0]) != 0:
                    pass
            except ValueError:
                pre.pop()  # 程式到這邊後，應該要接著比對strs[i]，但是現在的寫法會接著處理下個字串strs[i+1]
                if len(pre) == 0:
                    return ""
        return "".join(pre)


sol = Solution()
my_list = ["flower", "flow", "flight"]
print(sol.longestCommonPrefix(my_list))

File: 1-Easy/14. Longest Common Prefix/Horizontal_scanning.java
public class Horizontal_scanning {

    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0)
            return "";
        String prefix = strs[0];
        for (int i = 1; i < strs.length; i++)
            while (strs[i].indexOf(prefix) != 0) {
                prefix = prefix.substring(0, prefix.length() - 1);
                if (prefix.isEmpty())
                    return "";
            }
        return prefix;
    }

    public static void main(String[] args) {
        String[] strs = { "flower", "flow", "flight" };
        System.out.println(strs[1].indexOf(strs[0]));
        Horizontal_scanning horizontal_scanning = new Horizontal_scanning();
        System.out.println(horizontal_scanning.longestCommonPrefix(strs));
    }
}